### PR TITLE
Keep track of the final sync token in response

### LINKED ISSUE #266: Sync engine starts over from page 1 every time
In https://github.com/watermarkchurch/wcc-contentful/pull/257 we removed the direct linkage to the next page in the Sync response.  Unfortunately we had been using that to fetch the `next_sync_token` from the end of the response.

The current (broken) behavior is that even after iterating the sync response, the `next_sync_token` is the token from the first page only.  So, the next time that `SyncEngine#next` is called, it starts over from Page 1, then Page 2, then Page 3... rather than from the end of the list.

### PR DESCRIPTION
**Pull request by gburgett**
fixes #266

### CODE REVIEW

**Comment by gburgett** [comment]
I'm not 100% sure I like the approach on this one.  I'd like to talk it over with you @jpowell when you get back in the office.

**Comment by gburgett** [comment]
I like this approach better, "sync" in practice means "give me everything that changed" not "give me just the first page", and so the block-based syntax makes more sense to me.

File: wcc-contentful/lib/wcc/contentful/simple_client/response.rb [hunk 1]
```diff
@@ -129,26 +131,31 @@ def next_page
 
       next_page = SyncResponse.new(next_page)
       next_page.assert_ok!
+      @next_page = next_page if @memoize
```
Comment (gibsorya) [inline wcc-contentful/lib/wcc/contentful/simple_client/response.rb:134]: I'm guessing this allows the old functionality if they use the `sync_old` method?

**Review by gibsorya — APPROVED** [review]
Looks good to me!

File: wcc-contentful/lib/wcc/contentful/simple_client/response.rb (hunk 1 above)
Comment (gibsorya) [reply wcc-contentful/lib/wcc/contentful/simple_client/response.rb:134]: Not if they use the method directly, but if they don't provide a `block`

Comment (gburgett) [reply wcc-contentful/lib/wcc/contentful/simple_client/response.rb:134]: Correct

**Pull request merged**